### ISSUE TITLE
Type spec for `integer?` incorrect

### ISSUE THREAD

**Issue by imrekoszo** [clj-kondo/clj-kondo#2923]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

v2026.07.24

**problem**

clj-kondo thinks if `(integer? x)` returns truthy, `x` is of type `int`:

```shell
clj-kondo --lint - <<< "(fn [x] (when (integer? x) (int x)))"
<stdin>:1:28: info: Redundant int coercion: expression already has type int
linting took 122ms, errors: 0, warnings: 0

clj-kondo --lint - <<< "(fn [x] (when (integer? x) (long x)))"
linting took 60ms, errors: 0, warnings: 0
```

However, `integer?` returns true for several integer types, like Long, Short etc: https://github.com/clojure/clojure/blob/clojure-1.12.4/src/clj/clojure/core.clj#L1388

**example where removing the explicit int coercion changes program behavior**

Convoluted example, int coercions aren't too common in code I know, but they are usually there for a reason.

```clojure
(let [x 5]
  (when (integer? x)
    (= Integer (type (int x)))))
;=> true

(let [x 5]
  (when (integer? x)
    (= Integer (type x))))
;=> false
```
